filtering/filter.py

```diff
@@ -117,9 +117,9 @@ def prepare(source_file, target_file, source_lang, target_lang, lower=False):
     df_source = df["Source"]
     df_target = df["Target"]
 
-    df_source.to_csv(source_file, header=False, index=False, quoting=csv.QUOTE_NONE, sep="\n")
+    df_source.to_csv(source_file, header=False, index=False, quoting=csv.QUOTE_NONE, sep="\0")
     print("--- Source Saved:", source_file)
-    df_target.to_csv(target_file, header=False, index=False, quoting=csv.QUOTE_NONE, sep="\n")
+    df_target.to_csv(target_file, header=False, index=False, quoting=csv.QUOTE_NONE, sep="\0")
```

Comment:
This change causes `TypeError: "delimiter" must be a 1-character string`. It seems to me that it should remain `"\n"` when writing.

Reply:
I agree with @AaronCGoidel that "\0" can be used on in reading, but not for writing. Could you please update it @Nov00786 

For consistency, I think we should do the same for the two files under [train_dev_split](https://github.com/ymoslem/MT-Preparation/tree/main/train_dev_split)

Many thanks!

Reply:
updated

Reply:
@Nov00786  We agreed to keep it as "\0" in `pd.read_csv`, right?

Reply:
@ymoslem I apologize you are right it should be \0 in reading. thank you for correcting my mistake

Reply:
Thanks to you @Nov00786 and @AaronCGoidel!